### DISCUSSION TITLE
Add translations

### DISCUSSION THREAD

**Discussion by Alloyd031** [Sketchware-Pro/Sketchware-Pro#106] [Ideas]
Hi. I'd like to make a translations for this mod of Sketchware.

**Comment by j-px** [comment]
Sketchware already supports language packs.

**Comment by Alloyd031** [comment]
No, I want to translate a mod, not Sketchware.

**Comment by j-px** [comment]
What parts of the mod do you wanna translate?

**Comment by Alloyd031** [comment]
All mod

**Comment by j-px** [comment]
What part exactly? Most of the Mod uses Sketchware's strings, so it's not needed, but most Modder's classes use hardcoded strings, so translation will be hard, unless switching to resource strings

**Comment by Alloyd031** [comment]
I translated strings.xml, but it translates only Sketchware, not mod.

**Comment by j-px** [comment]
Well, if it's Sketchware's strings.xml only, upload it as Language pack in Sketchware instead, for a more consistent experience.

**Comment by Alloyd031** [comment]
I want to translate, for example, Blocks manager.

**Comment by j-px** [comment]
Alright then! I think that translations from strings.xml are best though, so we'll have to rewrite Modders' classes to use them, and then you can start writing translations for them! We'll notify you then

**Comment by Alloyd031** [comment]
Thank you!

**Comment by Alloyd031** [comment]
How long you rewrite Modders' classes?

**Comment by j-px** [comment]
I can't tell exactly when, but it's not too high on our to do list.

**Comment by ToastcodeDev** [comment]
May I ask about the percentage?

**Comment by ilyassesalama** [comment]
The percentage of what?
